### PR TITLE
Wrap stream from HadoopInputFile to work around #5201

### PR DESCRIPTION
**Pull request by phd3**
https://github.com/trinodb/trino/issues/5201#issuecomment-876583249

### CODE REVIEW

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/HdfsInputFile.java [hunk 1]
```diff
@@ -80,4 +81,48 @@ public String toString()
                 .add("user", user)
                 .toString();
     }
+
+    private static SeekableInputStream wrap(SeekableInputStream stream)
```
Comment (electrum) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/HdfsInputFile.java:85]: Nit: could inline this method (we mention wrapping in the comment)

**Review by electrum — APPROVED** [review]

**Comment by phd3** [comment]
Unrelated #8477

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/HdfsInputFile.java [hunk 2]
```diff
@@ -57,7 +57,8 @@ public long getLength()
     @Override
     public SeekableInputStream newStream()
     {
-        return environment.doAs(user, delegate::newStream);
+        // Hack: this wrapping is required to circumvent https://github.com/trinodb/trino/issues/5201
+        return new HdfsInputStream(environment.doAs(user, delegate::newStream));
```
Comment (findepi) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/HdfsInputFile.java:61]: How does wrapping help alleviate #5201? what's the mechanics?

Comment (phd3) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/HdfsInputFile.java:61]: @findepi does this help with mechanics? https://github.com/trinodb/trino/issues/5201#issuecomment-876583249

Comment (phd3) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/HdfsInputFile.java:61]: i.e. the "rewrapping" is not done. https://github.com/apache/iceberg/blob/master/parquet/src/main/java/org/apache/iceberg/parquet/ParquetIO.java#L93

**Pull request merged**